File: src/pocketcloud/console/log/level/CloudLogLevel.php
<?php

namespace pocketcloud\console\log\level;

use pocketcloud\util\EnumTrait;

/**
 * @method static CloudLogLevel INFO()
 * @method static CloudLogLevel WARN()
 * @method static CloudLogLevel ERROR()
 * @method static CloudLogLevel DEBUG()
 */
final class CloudLogLevel {
    use EnumTrait;

    protected static function init(): void {
        self::register("info", new CloudLogLevel("INFO", "§bINFO"));
        self::register("warn", new CloudLogLevel("WARN", "§cWARN"));
        self::register("error", new CloudLogLevel("ERROR", "§4ERROR"));
        self::register("debug", new CloudLogLevel("debug", "§6DEBUG"));
    }

    public function __construct(private string $name, private string $prefix) {}

    public function getName(): string {
        return $this->name;
    }

    public function getPrefix(): string {
        return $this->prefix;
    }
}

File: cloud-api/src/main/java/de/pocketcloud/api/logging/CloudLogLevel.java
package de.pocketcloud.api.logging;

import lombok.Getter;
import lombok.experimental.Accessors;

@Getter
@Accessors(fluent = true)
public enum CloudLogLevel {

    INFO("§bINFO"),
    WARN("§cWARN"),
    ERROR("§4ERROR"),
    SUCCESS("§aSUCCESS"),
    DEBUG("§6DEBUG");

    public static final CloudLogLevel LONGEST = SUCCESS;

    private final String prefix;

    CloudLogLevel(String prefix) {
        this.prefix = prefix;
    }

    public static String padPrefixToLength(CloudLogLevel level) {
        return " ".repeat(Math.max(0, (LONGEST.prefix.length() - level.prefix.length()))) + level.prefix;
    }
}